warehouse/query-core/src/main/java/datawave/query/data/parsers/TermFrequencyKey.java

```diff
@@ -0,0 +1,160 @@
+ * <li>column qualifier - "datatype\0uid\value\0FIELD</li>
+ * </ul>
+ */
+public class TermFrequencyKey implements KeyParser {
+    
+    // CQ partitions
+    private String datatype;
+    private String uid;
+    private String value;
+    private String field;
+    private String uidAndValue;
+    
+    private ByteSequence cq;
+    
+    private int firstNull;
+    private int secondNull;
+    private int thirdNull;
+    
+    private Key key;
+    
+    @Override
+    public void parse(Key k) {
+        clearState();
+        this.key = k;
+    }
+    
+    @Override
+    public void clearState() {
+        this.key = null;
+        
+        this.datatype = null;
+        this.uid = null;
+        this.value = null;
+        this.field = null;
+        
+        this.uidAndValue = null;
+        this.cq = null;
+        
+        this.firstNull = -1;
+        this.secondNull = -1;
+        this.thirdNull = -1;
+    }
+    
+    /**
+     * Iterate to find the split points
+     */
+    public void traverseColumnQualifier() {
+        
+        // bail out if the backing key is null, or if any split point was found
+        if (key == null || firstNull != -1 || secondNull != -1 || thirdNull != -1) {
+            return;
+        }
+        
+        this.cq = key.getColumnQualifierData();
+        
+        // Find all possible split points
+        for (int i = 0; i < cq.length(); i++) {
+            if (cq.byteAt(i) == 0x00) {
+                if (firstNull == -1) {
+                    firstNull = i;
+                } else {
+                    secondNull = i;
+                    break;
+                }
+            }
+        }
+        
+        // find third null via reverse iteration
+        for (int i = cq.length() - 1; i >= 0; i--) {
+            if (cq.byteAt(i) == 0x00) {
+                thirdNull = i;
+                break;
+            }
+        }
+    }
+    
+    @Override
+    public String getDatatype() {
+        if (datatype == null) {
+            traverseColumnQualifier();
+            
+            if (firstNull != -1 && cq != null) {
+                datatype = cq.subSequence(0, firstNull).toString();
+            } else {
+                throw new IllegalArgumentException("Failed to parse DATATYPE from tf key");
+            }
+        }
+        return datatype;
+    }
+    
+    @Override
+    public String getUid() {
+        if (uid == null) {
+            traverseColumnQualifier();
+            
+            if (firstNull != -1 && secondNull != -1 && cq != null) {
+                uid = cq.subSequence(firstNull + 1, secondNull).toString();
+            } else {
+                throw new IllegalArgumentException("Failed to parse UID from tf key");
+            }
+        }
+        return uid;
+    }
+    
+    @Override
+    public String getRootUid() {
+        if (uid == null) {
+            throw new IllegalArgumentException("Failed to parse root UID from tf key");
+        }
+        return null;
```

Comment:
this doesn't look right

Reply:
Hadn't implemented that yet for the TF key parser. Should be added now.